src/cache_cf.cc

```diff
@@ -620,18 +628,28 @@ parseConfigFileOrThrow(const char *file_name)
     return err_count;
 }
 
-// TODO: Refactor main.cc to centrally handle (and report) all exceptions.
-int
+void
 parseConfigFile(const char *file_name)
 {
+    static const char *kparseBlurb = "Log lines above or 'squid -k parse' may have more details.";
     try {
-        return parseConfigFileOrThrow(file_name);
-    }
-    catch (const std::exception &ex) {
-        debugs(3, DBG_CRITICAL, "FATAL: bad configuration: " << ex.what());
-        self_destruct();
-        return 1; // not reached
+        if (const auto err_count = parseConfigFileOrThrow(file_name)) {
+            // TODO: rollback to previous working settings
+            // for now any errors are a fatal condition...
+            const auto errMsg = ToSBuf("Found ", err_count, " configuration issues");
+            if (opt_parse_cfg_only)
+                Mem::Report();
+            throw TextException(errMsg, Here());
+        }
+
+    } catch (...) {
```

Comment:
It looks like something went wrong here: parseConfigFileOrThrow() returns an error count that nobody here really needs and that this code has to convert into an exception right after calling the function designed to throw exceptions on errors. Please refactor to throw inside parseConfigFileOrThrow() if needed. parseConfigFileOrThrow() should return void.